commit 90a4b64134ce0cb9a31cdceb49e265e8e2f88b56
Author: Álvaro Herrera <alvherre@alvh.no-ip.org>
Date:   Fri Aug 5 11:55:52 2022 +0200

    regress: fix test instability
    
    Having additional triggers in a test table made the ORDER BY clauses in
    old queries underspecified.  Add another column there for stability.
    
    Per sporadic buildfarm pink.

diff --git a/src/test/regress/expected/triggers.out b/src/test/regress/expected/triggers.out
index f131405bc7a..8fe42ec5604 100644
--- a/src/test/regress/expected/triggers.out
+++ b/src/test/regress/expected/triggers.out
@@ -2683,11 +2683,11 @@ create trigger tg after insert on parent
   for each row execute procedure trig_nothing();
 create trigger tg_stmt after insert on parent
   for statement execute procedure trig_nothing();
 select tgrelid::regclass, tgname, tgenabled from pg_trigger
   where tgrelid in ('parent'::regclass, 'child1'::regclass)
-  order by tgrelid::regclass::text;
+  order by tgrelid::regclass::text, tgname;
  tgrelid | tgname  | tgenabled 
 ---------+---------+-----------
  child1  | tg      | O
  parent  | tg      | O
  parent  | tg_stmt | O
@@ -2695,11 +2695,11 @@ select tgrelid::regclass, tgname, tgenabled from pg_trigger
 
 alter table only parent enable always trigger tg;	-- no recursion because ONLY
 alter table parent enable always trigger tg_stmt;	-- no recursion because statement trigger
 select tgrelid::regclass, tgname, tgenabled from pg_trigger
   where tgrelid in ('parent'::regclass, 'child1'::regclass)
-  order by tgrelid::regclass::text;
+  order by tgrelid::regclass::text, tgname;
  tgrelid | tgname  | tgenabled 
 ---------+---------+-----------
  child1  | tg      | O
  parent  | tg      | A
  parent  | tg_stmt | A
@@ -2708,11 +2708,11 @@ select tgrelid::regclass, tgname, tgenabled from pg_trigger
 -- The following is a no-op for the parent trigger but not so
 -- for the child trigger, so recursion should be applied.
 alter table parent enable always trigger tg;
 select tgrelid::regclass, tgname, tgenabled from pg_trigger
   where tgrelid in ('parent'::regclass, 'child1'::regclass)
-  order by tgrelid::regclass::text;
+  order by tgrelid::regclass::text, tgname;
  tgrelid | tgname  | tgenabled 
 ---------+---------+-----------
  child1  | tg      | A
  parent  | tg      | A
  parent  | tg_stmt | A
diff --git a/src/test/regress/sql/triggers.sql b/src/test/regress/sql/triggers.sql
index cb6fc4a90ee..9d21bbe2a94 100644
--- a/src/test/regress/sql/triggers.sql
+++ b/src/test/regress/sql/triggers.sql
@@ -1867,22 +1867,22 @@ create trigger tg after insert on parent
   for each row execute procedure trig_nothing();
 create trigger tg_stmt after insert on parent
   for statement execute procedure trig_nothing();
 select tgrelid::regclass, tgname, tgenabled from pg_trigger
   where tgrelid in ('parent'::regclass, 'child1'::regclass)
-  order by tgrelid::regclass::text;
+  order by tgrelid::regclass::text, tgname;
 alter table only parent enable always trigger tg;	-- no recursion because ONLY
 alter table parent enable always trigger tg_stmt;	-- no recursion because statement trigger
 select tgrelid::regclass, tgname, tgenabled from pg_trigger
   where tgrelid in ('parent'::regclass, 'child1'::regclass)
-  order by tgrelid::regclass::text;
+  order by tgrelid::regclass::text, tgname;
 -- The following is a no-op for the parent trigger but not so
 -- for the child trigger, so recursion should be applied.
 alter table parent enable always trigger tg;
 select tgrelid::regclass, tgname, tgenabled from pg_trigger
   where tgrelid in ('parent'::regclass, 'child1'::regclass)
-  order by tgrelid::regclass::text;
+  order by tgrelid::regclass::text, tgname;
 drop table parent, child1;
 
 -- Verify that firing state propagates correctly on creation, too
 CREATE TABLE trgfire (i int) PARTITION BY RANGE (i);
 CREATE TABLE trgfire1 PARTITION OF trgfire FOR VALUES FROM (1) TO (10);
